services/provider/server/server.go

```diff
@@ -105,26 +107,57 @@ func NewOCSProviderServer(ctx context.Context, namespace string) (*OCSProviderSe
 		return nil, fmt.Errorf("failed to create new scheme. %v", err)
 	}
 
-	client, err := util.NewK8sClient(scheme)
+	config, err := config.GetConfig()
 	if err != nil {
-		return nil, fmt.Errorf("failed to create new client. %v", err)
+		return nil, err
 	}
 
-	consumerManager, err := newConsumerManager(ctx, client, namespace)
+	cache, err := cache.New(config, cache.Options{
+		Scheme: scheme,
+		DefaultNamespaces: map[string]cache.Config{
+			namespace: {},
+		},
+	})
 	if err != nil {
-		return nil, fmt.Errorf("failed to create new OCSConumer instance. %v", err)
+		return nil, err
+	}
+
+	if err := cache.IndexField(
+		ctx,
+		&ocsv1alpha1.StorageConsumer{},
+		util.ObjectUidIndexName,
+		util.ObjectUidIndexFieldFunc,
+	); err != nil {
+		return nil, fmt.Errorf("failed to set up FieldIndexer on StorageConsumer for uid: %v", err)
 	}
```

Comment:
The index itself is an assumption of the consumer manager and is only consumed (and required) by it. It does not make sense to set it from the outside.